### PR TITLE
Fix setBallState in Er Force Simulator

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
Fix setBallState in Er Force Simulator and added tests. Previously, setBallState didn't do anything

### Testing Done
added unit tests

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/extlibs/er_force_sim/src/amun/simulator/simulator.h [hunk 1]
```diff
@@ -106,14 +134,28 @@ class camun::simulator::Simulator : public QObject
...
      */
     world::SimulatorState getSimulatorState();
-    void handleSimulatorSetupCommand(const std::shared_ptr<amun::Command> &command);
+
+    /**
+     * Handles a simulator set up command and configure the simulator accordingly
+     *
+     * @param command the simulator set up command
+     */
+    void handleSimulatorSetupCommand(const std::unique_ptr<amun::Command> &command);
```
Comment (Joshuayellowley) [inline src/extlibs/er_force_sim/src/amun/simulator/simulator.h:143]: No issues, but I was just wondering why the change here was made to swap from shared_ptr to unique_ptr?  Is there an advantage to using a unique_ptr here instead of a shared_ptr?

**Review by Joshuayellowley — DISMISSED** [review]
LGTM!

File: src/extlibs/er_force_sim/src/amun/simulator/simulator.h (hunk 1 above)
Comment (jonathanlew) [reply src/extlibs/er_force_sim/src/amun/simulator/simulator.h:143]: yeah unique_ptrs limit where the data can be accessed. It's a code quality thing that we'll have a presentation on at some point

File: src/software/simulation/firmware_object_deleter.h [hunk 2]
```diff
@@ -39,6 +39,7 @@ struct FirmwareRobotDeleter
         delete controller_state;
 
         app_firmware_robot_force_wheels_destroy(firmware_robot);
+        app_firmware_robot_velocity_wheels_destroy(firmware_robot);
```
Comment (akhilveeraghanta) [inline src/software/simulation/firmware_object_deleter.h:42]: is this change still required?

**Review by akhilveeraghanta — APPROVED** [review]
overall lgtm, just 1 non blocking comment

File: src/software/simulation/firmware_object_deleter.h (hunk 2 above)
Comment (jonathanlew) [reply src/software/simulation/firmware_object_deleter.h:42]: there was a memory leak, this fixes it

**Review by Joshuayellowley — APPROVED** [review]

**Review by xyztiger — APPROVED** [review]
